tests/lsdb/catalog/test_polygon_search.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from spherical_geometry.polygon import SingleSphericalPolygon
+
+
+def test_polygon_search_filters_correct_points(small_sky_order1_catalog):
+    ra, dec = [300, 300, 272, 272], [-50, -55, -55, -50]
+    polygon = SingleSphericalPolygon.from_lonlat(ra, dec)
+
+    polygon_search_df = small_sky_order1_catalog.polygon_search(polygon).compute()
+
+    for _, row in polygon_search_df.iterrows():
+        ra = row[small_sky_order1_catalog.hc_structure.catalog_info.ra_column]
+        dec = row[small_sky_order1_catalog.hc_structure.catalog_info.dec_column]
+        assert polygon.contains_lonlat(ra, dec)
+
+
+def test_polygon_search_filters_partitions(small_sky_order1_catalog):
+    ra, dec = [300, 300, 272, 272], [-50, -55, -55, -50]
+    polygon = SingleSphericalPolygon.from_lonlat(ra, dec)
+
+    hc_polygon_search = small_sky_order1_catalog.hc_structure.filter_by_polygon(polygon)
+    polygon_search_catalog = small_sky_order1_catalog.polygon_search(polygon)
+
+    assert len(hc_polygon_search.get_healpix_pixels()) == len(polygon_search_catalog.get_healpix_pixels())
+    assert len(hc_polygon_search.get_healpix_pixels()) == polygon_search_catalog._ddf.npartitions
+    for pixel in hc_polygon_search.get_healpix_pixels():
+        assert pixel in polygon_search_catalog._ddf_pixel_map
+
+
+def test_polygon_search_empty(small_sky_order1_catalog):
+    ra, dec = [0, 1, 0], [0, 1, 2]
+    polygon = SingleSphericalPolygon.from_lonlat(ra, dec)
+    polygon_search_catalog = small_sky_order1_catalog.polygon_search(polygon)
+    assert len(polygon_search_catalog.get_healpix_pixels()) == 0
+    assert len(polygon_search_catalog.hc_structure.pixel_tree) == 1
+
+
+def test_polygon_search_invalid_shape(small_sky_order1_catalog):
+    # Polygon is not convex, so the shape is invalid
+    ra, dec = [0, 1, 1, 0], [1, 0, 1, 0]
+    polygon = SingleSphericalPolygon.from_lonlat(ra, dec)
+    with pytest.raises(RuntimeError):
```

Comment:
This is a runtime error? Are there any more details in the error message?

Reply:
The exception thrown is a `RuntimeError('Unknown exception')`, with no additional data. There is some output log stating that the "polygon is not convex" but it seems like it's being thrown in C code ([`healpix_base.cc/query_polygon_internal`](https://github.com/healpy/healpixmirror/blob/bc5cbaced09007fdda7f1fd9ea953d57dae7d2c5/src/cxx/Healpix_cxx/healpix_base.cc#L984C27-L984C49)) and not being sent up the chain.